options/options.js

```diff
@@ -16,21 +16,105 @@ function saveOptions(e) {
 function restoreOptions() {
   var value = '';
   var getting = '';
-  document.querySelectorAll('input[id]').forEach(function(item) {
-    item = item.id;
+  document.querySelectorAll('input[id]').forEach(function(el) {
+    var item = el.id;
     getting = browser.storage.local.get(item);
     getting.then(function(result) {
       value = result[Object.keys(result)[0]];
-      if (Number.isInteger(parseInt(value)) || document.querySelector('#' + item).type === 'url') {
-		if(value !== undefined) {
-		  document.querySelector('#' + item).value = value;
-		}
+      var isInt = Number.isInteger(parseInt(value, 10));
+      // Why not using the argument of forEach here instead of querying the DOM?
```

Comment:
let's do it your way, my JS skills are lacking sometimes 😂

Reply:
Roger that. Letme refactor it while I'm at it!

Reply:
Okay, left some comments for you to learn a bit while reviewing :+1: